File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/C_Remainder_Minimization_2019.cpp
#include <bits/stdc++.h>
#define INF 1'000'000'000
using namespace std;
typedef long long ll;

ll L, R;

void solve() {
  ll res = INF;
  if (R - L >= 2019) {
    res = 0;
  } else {
    for (ll i = L; i < R; ++i) {
      for (ll j = i + 1; j <= R; ++j) {
        res = min(res, i % 2019 * j % 2019 % 2019);
      }
    }
  }
  printf("%lld\n", res);
}

int main() {
  cin >> L >> R;
  solve();
}

File: AtCoder/ACP/BootCamp4b/Medium100_2/python/C_Remainder_Minimization_2019.py
INF = float('inf')
L, R = map(int, input().split())
res = INF
if R - L >= 2019:
    res = 0
else:
    for i in range(L, R):
        for j in range(i + 1, R + 1):
            res = min(res, i % 2019 * j % 2019 % 2019)
print(res)
